### DISCUSSION TITLE
statamic:assets:generate-presets fails

### DISCUSSION THREAD

**Discussion by eliasu** [statamic/cms#5634] [Troubleshooting] [answered]
Hey there, 

I have this Problem: We are deploying a Statamic Site via Ploi.io. So far everything works fine, but today the deploy failed due to this deploy-script-command:  `statamic:assets:generate-presets`. I tested it without, to be sure it's this command. As I understand it, this is the command for creating image variants (btw: I use this addon [responsive images](https://statamic.com/addons/spatie/responsive-images) for creating my responsive images). 

This is the error from the log:

```
[✗] No user defined presets.

[  1/214] Generating cp_thumbnail_small_landscape... 360_F_222851624_jfoMGbJxwRi5AWGdPgXKSABMnzCQo9RN.jpg
[ 86/214] Generating cp_thumbnail_small_landscape... zoom_berenberg_book_4004bb7c6e0f.jpg
   Intervention\Image\Exception\NotReadableException 

  Unsupported image type image/tiff. GD driver is only able to decode JPG, PNG, GIF, BMP or WebP files.

  at vendor/intervention/image/src/Intervention/Image/Gd/Decoder.php:75
     71▕                 $core = @imagecreatefrombmp($path);
     72▕                 break;
     73▕ 
     74▕             default:
  ➜  75▕                 throw new NotReadableException(
     76▕                     sprintf("Unsupported image type %s. GD driver is only able to decode JPG, PNG, GIF, BMP or WebP files.", strtolower($mime))
     77▕                 );
     78▕         }
     79▕

      +45 vendor frames 
  46  artisan:37
      Illuminate\Foundation\Console\Kernel::handle()
```

So, it seems, the conversion is not possible for `.tiff` images. And it's true, my client tried to use a `.tiff` image today. So I deleted the image but the error persisted. 

Is there any command for clearing the cache (of course I did clear the cache, cleared glide, etc) with some command I don't know? Or where could I look for the `.tiff` image manually on the server to delete it? Or am I on the wrong track?

Any help would be great!
Thanks.

**Comment by afonic** [comment]
If you deleted the file, maybe try `php please assets:meta`, in case the .yaml file remained in the system. Or search for it in the .meta folder in the folder the file was in and delete it manually. (it's hidden in Mac / Linux)

You could also try installing imagick and changing the setting in `config/statamic/assets.php`.

However this seems like a bug in the generate-presets command to be honest, it should probably handle this exception and move on.

**Reply by eliasu** [reply]
thanks for your reply. Actually I already tried `assets:meta` a couple of times … and `glide:clear` but nothing really helps. 
I searched the server for `"*tiff"` (also hidden folders), looked in the .meta folders – nothing there. And the error message is not really telling me which file is corrupted. 

> You could also try installing imagick and changing the setting in config/statamic/assets.php.

What do you mean by that. I could install imagick, but how to edit the config? And what after that? How does that help?


Any other ideas?

**Reply by afonic** [reply]
The GD driver cannot handle TIFF files. You can install Imagick and change the driver in `config/statamic/assets.php` (line 42).

You can Google how to install Imagick for PHP depending on your system.

Having said that, maybe try clearing folder by folder and see where the file is? It may be that you have a .jpg named file but it's really tiff. (some users change the extension and think that they've changed the file)

**Reply by jasonvarga** [reply]
The output shows shows `zoom_berenberg_book_4004bb7c6e0f.jpg` - Maybe check that file.

**Comment by eliasu** [comment, marked as answer]
Thank you all for your help. I figured it out. Solution below for people with the same issues. 

> It may be that you have a .jpg named file but it's really tiff.
I was hoping to find the error elsewhere but no: That was the reason. They "converted" a `.tiff` to a `.jpg` image via renaming 😔

Since I had a lot of images, and couldn't just delete everything, or go through every file. 

So I use the command line with a relatively simple command to check the "real file type" behind the naming suffix. Then you get a long list, which you can copy to a text editor and search for all unwanted types. This way I got the corrupted files. Maybe it helps someone as well. 

```
for file in /path/to/your/folder/*
do
    file -I  "$file"
done
```

Thanks again. Closed!
